platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -154,6 +156,8 @@ public Object saveInstance(EntityRecord entityRecord, Long deleteValueFieldId) {
             } else {
                 return service.update(instance);
             }
+        } catch (JDODataStoreException e) {
+            throw new DuplicateEntryException(entity.getName(), e);
```

Comment:
We don't want "DuplicateEntryException" to be thrown in case of all datastore exceptions, only in case of unique violations